src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -372,6 +373,20 @@ private void OnHostInitializing(object sender, EventArgs e)
             IsHostHealthy(throwWhenUnhealthy: true);
         }
 
+        private void OnHostInitialized(object sender, EventArgs e)
+        {
+            OnHostInitialized();
+        }
+
+        /// <summary>
+        /// Called after the host has been fully initialized, but before it
+        /// has been started.
+        /// </summary>
+        protected virtual void OnHostInitialized()
```

Comment:
Do we need this extra method here? This could be added at a later point if needed, but if we're not overriding this anywhere, let's just handle the state change in the private handler.